tests/dtm/test_single_process_restart.py

```diff
@@ -150,24 +180,162 @@ def test_write_during_m0d_restart(self):
+        responses = dict()
+        while len(responses) != 2:
+            responses = output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        pass_logs = list(x[1] for x in responses["pass_res"])
+        self.log.debug("Pass logs list: %s", pass_logs)
+        fail_logs = list(x[1] for x in responses["fail_res"])
+        self.log.debug("Fail logs list: %s", fail_logs)
+        resp = self.ha_obj.check_s3bench_log(file_paths=pass_logs)
+        assert_utils.assert_false(len(resp[1]), "Failed WRITEs/READs in background before/after m0d"
+                                                f"restart. Logs which contain failures: {resp[1]}")
+        resp = self.ha_obj.check_s3bench_log(file_paths=fail_logs, pass_logs=False)
+        assert_utils.assert_false(len(resp[1]) <= len(fail_logs),
+                                  "Expected failures in WRITEs/READs in background during m0d "
+                                  f"restart. Logs collected during m0d restart: {resp[1]}")
+        self.log.info("Step 3: Successfully completed WRITEs/READs in background")
+
+        self.log.info("Step 4: Perform READs-Verify on already written data in Step 1")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=self.iam_user,
+                                                    log_prefix=test_prefix, skipwrite=True,
+                                                    skipcleanup=True, nclients=10, nsamples=30)
+        assert_utils.assert_true(resp[0], resp[1])
+        self.log.info("Step 4: Successfully performed READs-Verify on already written data in "
+                      "Step 1")
+
+        self.log.info("ENDED: Verify IOs before and after RC pod m0d restart using pkill")
+
+    @pytest.mark.lc
+    @pytest.mark.dtm0
+    @pytest.mark.tags("TEST-41235")
+    def test_bkt_creation_ios_after_m0d_restart(self):
+        """Verify bucket creation and IOs after m0d restart using pkill."""
+        self.log.info("STARTED: Verify bucket creation and IOs after m0d restart using pkill")
+
+        self.log.info("Step 1: Perform Single m0d Process Restart")
+        self.log.debug("Get IDs of all m0d processes")
+        resp, fids = self.health_obj.hctl_status_get_svc_fids()
+        assert_utils.assert_true(resp, "Failed to get services fids")
+        fids = fids['ioservice']
+        resp = self.dtm_obj.process_restart(master_node=self.master_node_list[0],
+                                            health_obj=self.health_obj, pod_prefix=POD_NAME_PREFIX,
+                                            container_prefix=MOTR_CONTAINER_PREFIX,
+                                            process=self.m0d_process, process_ids=fids)
+        assert_utils.assert_true(resp, f"Response: {resp} \nhctl status is not as expected")
+        self.log.info("Step 1: m0d restarted and recovered successfully")
+
+        self.log.info("Step 2: Create IAM user and perform WRITEs/READs-Verify/DELETEs with "
```

Comment:
@ApurwaMohite  As discussed, IAM user creation/deletion wont be supported post m0d restart.

Reply:
As per discussion, creation should work even after m0d restart right? Deletion will have problems. Please confirm once.

Reply:
IAM user creation/deletion is not supported post m0d restart. Creation of IAM user was planned before m0d restart, hence it should work.